### PR TITLE
Fixed DatetimeIndex.tz_convert() .freq attribute

### LINKED ISSUE #49696: BUG: `DatetimeIndex.tz_convert()` has incorrcect `.freq` attribute
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [ ] I have confirmed this bug exists on the main branch of pandas.

### Reproducible Example

```python
import pandas as pd

# DST-transition. Frequency is days, because all timestamps are at same time - even if not all days span 24h. So far, so good.
i = pd.date_range('2020-03-27 06:00', freq='D', periods=5, tz='Europe/Berlin')
# DatetimeIndex(['2020-03-27 06:00:00+01:00', '2020-03-28 06:00:00+01:00',
#                '2020-03-29 06:00:00+02:00', '2020-03-30 06:00:00+02:00',
#                '2020-03-31 06:00:00+02:00'],
#               dtype='datetime64[ns, Europe/Berlin]', freq='D')

# Converting to other timezone gives timestamps with distinct times. This index should no longer have the daily frequency. However: it is initially given this frequency nonetheless (=Bug).
i2 = i.tz_convert('UTC')
# DatetimeIndex(['2020-03-27 05:00:00+00:00', '2020-03-28 05:00:00+00:00',
#                '2020-03-29 04:00:00+00:00', '2020-03-30 04:00:00+00:00',
#                '2020-03-31 04:00:00+00:00'],
#               dtype='datetime64[ns, UTC]', freq='D')

# We can confirm that this is incorrect, by manually trying to set the `.freq` attribute:
i2.freq = i2.freq #ValueError: Inferred frequency None from passed values does not conform to passed frequency D
```

### Issue Description

See inline comments.

### Expected Behavior

`i2` should not have a frequency attribute set.

### Installed Versions

<details>

INSTALLED VERSIONS
------------------
commit           : 91111fd99898d9dcaa6bf6bedb662db4108da6e6
python           : 3.9.13.final.0
python-bits      : 64
OS               : Windows
OS-release       : 10
Version          : 10.0.19045
machine          : AMD64
processor        : Intel64 Family 6 Model 165 Stepping 2, GenuineIntel
byteorder        : little
LC_ALL           : None
LANG             : None
LOCALE           : de_DE.cp1252

pandas           : 1.5.1
numpy            : 1.23.4
pytz             : 2022.6
dateutil         : 2.8.2
setuptools       : 63.4.1
pip              : 22.2.2
Cython           : None
pytest           : 7.1.3
hypothesis       : None
sphinx           : 5.3.0
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : None
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : 3.1.2
IPython          : 8.5.0
pandas_datareader: None
bs4              : 4.11.1
bottleneck       : None
brotli           : None
fastparquet      : None
fsspec           : None
gcsfs            : None
matplotlib       : 3.6.2
numba            : None
numexpr          : None
odfpy            : None
openpyxl         : 3.0.10
pandas_gbq       : None
pyarrow          : None
pyreadstat       : None
pyxlsb           : None
s3fs             : None
scipy            : None
snappy           : None
sqlalchemy       : None
tables           : None
tabulate         : None
xarray           : None
xlrd             : None
xlwt             : None
zstandard        : None
tzdata           : None

</details>

### PR DESCRIPTION
**Pull request by Detoro**
- [ ] closes #49696
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) 
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).

### CODE REVIEW

**Comment by Detoro** [comment]
pre-commit.ci autofix

File: pandas/core/arrays/datetimes.py [hunk 1]
```diff
@@ -896,7 +897,7 @@ def tz_convert(self, tz) -> Self:
 
         # No conversion since timestamps are all UTC to begin with
         dtype = tz_to_dtype(tz, unit=self.unit)
-        return self._simple_new(self._ndarray, dtype=dtype, freq=self.freq)
```
Comment (mroeschke) [inline pandas/core/arrays/datetimes.py:899]: I don't think this is an appropriate fix

**Review by mroeschke — CHANGES_REQUESTED** [review]

File: pandas/core/arrays/datetimes.py (hunk 1 above)
Comment (Detoro) [reply pandas/core/arrays/datetimes.py:899]: Why do you think so? and what would be an appropriate fix?

Comment (mroeschke) [reply pandas/core/arrays/datetimes.py:899]: Based on the failing tests seems like this behavior was relied upon

Comment (Detoro) [reply pandas/core/arrays/datetimes.py:899]: Resolving this issue is undoing work done a while ago. The frequency attribute was set as part of this issue [#25241 (https://github.com/pandas-dev/pandas/issues/25241) - why do you think it should not be reset?

Comment (mroeschke) [reply pandas/core/arrays/datetimes.py:899]: Based on the failing unit test the freq should be preserved

**Comment by mroeschke** [comment]
Thanks for the pull request, but it appears to have gone stale. If interested in continuing, please merge in the main branch, address any review comments and/or failing tests, and we can reopen.

**Pull request closed without merging**